client/src/Utils/ControllerHooks/useExposuresSaving.ts

```diff
@@ -6,18 +6,36 @@ import {fieldsNames, ExposureAndFlightsDetailsAndSet, Exposure } from 'commons/C
 import axios from '../axios';
 import useDBParser from '../vendor/useDBParsing';
 
+const exposureDeleteCondition = 
+    (wereFlights: boolean, wereConfirmedExposures: boolean) : (exposure: Exposure) => boolean => {
+    if (!wereConfirmedExposures) return (exposure: Exposure) => exposure.wasConfirmedExposure;
+    if (!wereFlights) return (exposure: Exposure) => exposure.wasAbroad;
+    return (exposure: Exposure) => false;
+}
+
+
 const useExposuresSaving = (exposuresAndFlightsVariables: ExposureAndFlightsDetailsAndSet) => {
     const epidemiologyNumber = useSelector<StoreStateType, number>(state => state.investigation.epidemiologyNumber);
     const {parseLocation} = useDBParser();
 
     const saveExposureAndFlightData = async () : Promise<void> => {
-        let { exposures } = exposuresAndFlightsVariables.exposureAndFlightsData;
-        const exposuresPromises = exposures.map(async exposure => await extractExposureData(exposure));
-        exposures = await Promise.all(exposuresPromises);
+        const { exposures, wereFlights, wereConfirmedExposures, exposuresToDelete } = exposuresAndFlightsVariables.exposureAndFlightsData;
+        let filteredExposures : Exposure[] = [];
+        const filterCondition : (exposure: Exposure) => boolean = exposureDeleteCondition(wereFlights, wereConfirmedExposures);
+        exposures.forEach(exposure => {
+            if (filterCondition(exposure)) {
```

Comment:
why not (!wereConfirmedExposures && exposure.wasConfirmedExposure || !wereFlights&&exposure.wasAbroad) condition
instead of filterCondition and exposureDeleteCondition

Reply:
I think that less efficent to do because you will have to check wereConfirmedExposures every time

Reply:
but its much less code and much less functions call, i think that in overall the efficiency difference is minor

Reply:
I still think this will be much unnecessary condition checks